src/LTOCli/Commands/Sponsorship.py

```diff
@@ -2,32 +2,61 @@
 from LTO.Transactions.Sponsorship import Sponsorship
 from LTO.Transactions.CancelSponsorship import CancelSponsorship
 from LTOCli import Config
+from LTO.PublicNode import PublicNode
 
 def func(nameSpace,parser):
-    if nameSpace.option[0] == 'create':
+    if vars(nameSpace)['subparser-name-sponsorship'] == 'create':
+    #if nameSpace.option[0] == 'create':
         transaction = Sponsorship(nameSpace.recipient[0])
         transaction.signWith(handle.getAccount(parser))
         if vars(nameSpace)['unsigned'] is False:
             if vars(nameSpace)['account']:
-                transaction.signWith(handle.getAccountFromOption(parser, nameSpace['account'][0]))
+                transaction.signWith(handle.getAccountFromOption(parser, vars(nameSpace)['account'][0]))
             else:
                 transaction.signWith(handle.getAccount(parser))
             if vars(nameSpace)['no_broadcast'] is False:
                 transaction.broadcastTo(handle.getNode())
         elif vars(nameSpace)['no_broadcast'] is False:
             parser.error(
                 "Use the '--unsigned' option only in combination with the '--no-broadcast' option. Type 'lto sponsorship create --help' for more informations ")
-    else:
+    elif vars(nameSpace)['subparser-name-sponsorship'] == 'cancel':
         # cancel case
         transaction = CancelSponsorship(nameSpace.recipient[0])
         transaction.signWith(handle.getAccount(parser))
         if vars(nameSpace)['unsigned'] is False:
             if vars(nameSpace)['account']:
-                transaction.signWith(handle.getAccountFromOption(parser, nameSpace['account'][0]))
+                transaction.signWith(handle.getAccountFromOption(parser, vars(nameSpace)['account'][0]))
             else:
                 transaction.signWith(handle.getAccount(parser))
             if vars(nameSpace)['no_broadcast'] is False:
                 transaction.broadcastTo(handle.getNode())
         elif vars(nameSpace)['no_broadcast'] is False:
             parser.error(
-                "Use the '--unsigned' option only in combination with the '--no-broadcast' option. Type 'lto sponsorship cancel --help' for more informations ")
\ No newline at end of file
+                "Use the '--unsigned' option only in combination with the '--no-broadcast' option. Type 'lto sponsorship cancel --help' for more informations ")
+
+    elif vars(nameSpace)['subparser-name-sponsorship'] == 'list':
+        pass
+
+    elif vars(nameSpace)['subparser-name-sponsorship'] == 'list-inbound':
+        node = PublicNode('http://testnet.lto.network')
```

Comment:
I see this one getting initialised a lot of times, probably you can have it as an env var or something similar

Reply:
you're right, that should't be there but in the PublicNode class in the python-api which are currently under refactoring hence it's unfortunately there until the new version comes out